### PR TITLE
Update dot release process

### PR DESCRIPTION
**Pull request by lfbrock**

### CODE REVIEW

**Comment by lfbrock** [comment]
@jasonblais - I've just added the links to:
- Footer of mattermost.org
- Footer of mattermost.com
- https://about.mattermost.com/security-updates/

And I'll add to this PR to add it to: https://docs.mattermost.com/process/security.html

One thing I was thinking about though - we'll probably need to do two email versions for a release, one for the security update list and one for Mattermost insiders. Probably don't want to send the same email to both lists or people will get duplicates.

File: source/process/release-process.md [hunk 1]
```diff
@@ -258,6 +259,7 @@ Once final dot release build is ready to cut:
       - Test the download links before and after updating the page 
     - Update [Mattermost pricing page](https://about.mattermost.com/pricing/) if anything has changed
     - Add the download links to http://docs.mattermost.com/administration/upgrade.html#version-archive  
+    - Update [mattermost-docker](https://github.com/mattermost/mattermost-docker) version
```
Comment (esethna) [inline source/process/release-process.md]: Propose we link directly to the file, if this is it? https://github.com/mattermost/mattermost-docker/blob/master/app/Dockerfile

Comment (it33) [inline source/process/release-process.md]: We have draft update instructions here: https://github.com/mattermost/mattermost-docker/pull/91/files#diff-aa42d848e7e0ee6100ed717268346bd1R41

Propose opening a separate ticket to update this doc with a link after this is merged and the PR above is merged

**Comment by it33** [comment]
Great work! Looks good to merge, and we can add a ticket for the update link

**Pull request merged**